sklearn/pipeline.py

```diff
@@ -348,11 +349,14 @@ def fit(self, X, y=None, **fit_params):
         self : Pipeline
             This estimator
         """
-        Xt, fit_params = self._fit(X, y, **fit_params)
+        fit_params_steps = self._check_fit_params(**fit_params)
```

Comment:
You changed the order of some operations, e.g. `_validate_steps` is now performed after `_check_fit_params`.
Did you check that it does not matter ?

Reply:
Good point.

From what I can tell, `_validate_steps` has no effect on `_check_fit_params`.
And if `_validate_steps` fails on master, it will fail the same way in this PR.